### DISCUSSION TITLE
Criar seção da requisição

### DISCUSSION THREAD

**Discussion by rafapeiter** [HashLoad/horse#211] [Q&A]
Olá,

Teria como criar alguma identificação para cada requisição que é feita no servidor?
Sei que cada requisição é uma Thread, talvez usar o ID da Thread para identificar a requisição em si?

A ideia seria criar uma seção com variáveis que possam ser utilizadas em outros escopos do projeto.
Para isso precisaria de uma forma de identificar que é a seção da requisição X que está solicitando um determinado dado.

Seria algo como seção de usuário, porém neste caso é seção da requisição.

**Comment by Denis-Tek** [comment]
Você pode interceptar cada requisição/response e adicionar informações nos headeres.

Depois assista este vídeo pois deve abrir bastante a cabeça nas possibilidades do Horse: 

https://youtu.be/ZlmpOQdQjmI

A partir de 1:20:00h mostro um uso de interceptação das requisições que você pode utilizar.

**Comment by rafapeiter** [comment]
Na verdade preciso de uma identificação a nível de requisição só que no escopo que uso não tenho o header gerado pela requisição. 

Claro que poderia enviar como parâmetro para a função a própria requisição, neste caso eu teria a identificação. Nos lugares que preciso usar não tenho como passar como parâmetro.

Na função que uso não tenho os dados da requisição. Ex: 
 
```
   THorse.Get('ping',
    procedure(Req: THorseRequest; Res: THorseResponse; Next: TProc)
    begin
      GeraPedido;
      Res.Send('pong');
    end);

   procedure GeraPedido;
   begin
       // Aqui precisaria saber qual foi a requisição que chamou esta procedure.
   end;
```


O que pensei em fazer é usar o objeto da Thread que é gerada pela requisição do tipo "TIdThreadWithTask".
Posso gerar um identificador para essa Thread e usar na função GeraPedido.

Ex: 
```
    TThreadHorse = TIdThreadWithTask(class);
    private
      ID: integer;
    end;

    THorse.Get('ping',
    procedure(Req: THorseRequest; Res: THorseResponse; Next: TProc)
    begin
      TThreadHorse (TThread.CurrentThread).ID := xxxx; // Precisa criar uma classe que derive tela
      GeraPedido;
      Res.Send('pong');
    end);

   procedure GeraPedido;
   begin
      writeln(TThreadHorse (TThread.CurrentThread).ID);
   end;
```

Alguma outra sugestão? A classe do Horse possui algum outro objeto que poderia ser utilizado nesta situação?

**Reply by viniciussanchez** [reply]
Que eu me lembre, até o momento não tem nada referente a isso
